File: backend_book/OnlineBookstore.Api/OnlineBookstore.Domain/Entities/Order.cs
namespace OnlineBookstore.Domain.Entities
{
    public class Order
    {
        public long Id { get; set; }
        public long UserId { get; set; }
        public User User { get; set; } = default!;

        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = "Pending";
        public string ShippingAddress { get; set; } = default!;

        public ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();
    }

}

File: fontend_book/online-bookstore-client/src/types/Order.ts
import { OrderDetail } from "./OrderDetail";
import { User } from "./user";

export interface Order {
    id: number;
    userId: number;
    user: User;
    orderDate: Date;
    totalAmount: number;
    status: string;
    shippingAddress: string;
    orderDetails: OrderDetail[];
}